frappe/commands/utils.py

```diff
@@ -556,11 +556,28 @@ def run_ui_tests(context, app, headless=False):
 	site_env = 'CYPRESS_baseUrl={}'.format(site_url)
 	password_env = 'CYPRESS_adminPassword={}'.format(admin_password) if admin_password else ''
 
+	os.chdir(app_base_path)
+
+	node_bin = subprocess.getoutput("npm bin")
+	cypress_path = "{0}/cypress".format(node_bin)
+	plugin_path = "{0}/cypress-file-upload".format(node_bin)
+
+	# check if cypress in path...if not, install it.
+	if not (os.path.exists(cypress_path) or os.path.exists(plugin_path)):
+		# install cypress
+		click.secho("Installing Cypress...", fg="yellow")
+		frappe.commands.popen("yarn add cypress@3 cypress-file-upload@^3.1")
+
 	# run for headless mode
 	run_or_open = 'run --browser chrome --record --key 4a48f41c-11b3-425b-aa88-c58048fa69eb' if headless else 'open'
-	command = '{site_env} {password_env} yarn run cypress {run_or_open}'
-	formatted_command = command.format(site_env=site_env, password_env=password_env, run_or_open=run_or_open)
-	frappe.commands.popen(formatted_command, cwd=app_base_path, raise_err=True)
+	command = '{site_env} {password_env} {cypress} {run_or_open}'
+	formatted_command = command.format(site_env=site_env, password_env=password_env, cypress=cypress_path, run_or_open=run_or_open)
+	try:
+		click.secho("Running Cypress...", fg="yellow")
+		frappe.commands.popen(formatted_command, cwd=app_base_path, raise_err=True)
+	finally:
+		click.secho("Cleaning Up...", fg="yellow")
+		frappe.commands.popen("yarn remove cypress cypress-file-upload")
```

Comment:
This will also remove cypress from our dev setups?

Reply:
So, the idea is to remove cypress completely. Install it on the go only when someone wants to run UI tests

Reply:
> Maybe removing the installs is unnecessary

Yeah, we can remove it